nominal/sdk.py

```diff
@@ -210,24 +203,51 @@ def update(
             name=name,
             properties=None if properties is None else dict(properties),
         )
-        response = self._catalog_client.update_dataset_metadata(self._auth_header, self.rid, request)
+        response = self._client._catalog_client.update_dataset_metadata(self._client._auth_header, self.rid, request)
 
-        dataset = self.__class__._from_conjure(self._auth_header, self._catalog_client, response)
+        dataset = self.__class__._from_conjure(self._client, response)
         update_dataclass(self, dataset, fields=self.__dataclass_fields__)
         return self
 
+    def add_to_dataset_from_io(
+        self,
+        dataset: BinaryIO,
+        timestamp_column_name: str,
+        timestamp_column_type: TimestampColumnType,
+        file_extension: _AllowedFileExtensions = ".csv",
+    ) -> None:
+        """Append to a dataset from a file-like object."""
+
+        if isinstance(dataset, TextIOBase):
+            raise TypeError(f"dataset {dataset} must be open in binary mode, rather than text mode")
+
+        filename = f"{self.name}{file_extension}"
+        s3_path = put_multipart_upload(
+            self._client._auth_header, dataset, filename, "text/csv", self._client._upload_client
+        )
+        request = ingest_api.TriggerFileIngest(
+            destination=ingest_api.IngestDestination(
+                existing_dataset=ingest_api.ExistingDatasetIngestDestination(dataset_rid=self.rid)
+            ),
+            source=ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path)),
+            source_metadata=ingest_api.IngestSourceMetadata(
+                timestamp_metadata=ingest_api.TimestampMetadata(
+                    series_name=timestamp_column_name,
+                    timestamp_type=_timestamp_type_to_conjure_ingest_api(timestamp_column_type),
+                ),
+            ),
```

Comment:
can we add some quick validation to throw if it's not an absolute timestamp?

Reply:
realized that i forgot that as soon as I asked you to take a look lol